app/models/king.rb

```diff
@@ -1,2 +1,21 @@
 class King < Piece
-end
+	def valid_move?
+		# Find king's current position on board.
+		king = King.find_by(x, y)
+		# Check if valid move, or if obstructed.
+		if king.is_obstructed?
```

Comment:
Same for this. Since we are already inside the King object. We can call just `is_obstructed?` instead.